quickwit-core/src/index.rs

```diff
@@ -91,6 +93,9 @@ impl IndexService {
         &self,
         index_config: IndexConfig,
     ) -> Result<IndexMetadata, IndexServiceError> {
+        index_config
```

Comment:
Injecting the validation logic in `IndexService`. Ideally, we would validate identifiers in the metastore trait before creating objects.